### PR TITLE
refactor(test): add unit tests for agent/api

### PR DESCRIPTION
**Pull request by clebs**
- Add unit tests for the agent/api pkg.
- Fixed wrong argument order in JSON unit test.
- `util.RemoveJSONComment` now also removes trailing spaces and tabs from comments.

Part of #241

### CODE REVIEW

File: agent/internal/util/json.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 func RemoveJSONComment(strIn []byte) []byte {
 	str := string(strIn)
 	// edgecase that isn't handled: double quote in the comment
-	re1 := regexp.MustCompile(`//\s*?[^"\\S]*?\s*?(\n|\r|\r\n|\n\r)`)
+	re1 := regexp.MustCompile(`([ \t]*)//\s*?[^"\\S]*?\s*?(\n|\r|\r\n|\n\r)`)
```
Comment (nandor-magyar) [inline agent/internal/util/json.go:10]: \s includes tabs too.

> That is: \s matches a space, a tab, a carriage return, a line feed, or a form feed.
Source: https://www.regular-expressions.info/shorthand.html

Also, golang official: https://pkg.go.dev/regexp/syntax
```
\s             whitespace (== [\t\n\f\r ])
```

Is it necessary?

**Review by nandor-magyar — COMMENTED** [review]
Nice catch of the argument order in tests. :+1:

File: agent/internal/util/json.go (hunk 1 above)
Comment (clebs) [reply agent/internal/util/json.go]: You are right, `\s` is enough. Corrected the regex.

File: agent/api/v1/deploy_test.go [hunk 2]
```diff
@@ -27,3 +29,202 @@ func TestSetDeploymentDefaults(t *testing.T) {
...
+}
+
+func TestUnmarshalJSON(t *testing.T) {
+	j := v1.Base64JSONBytes{}
+
+	err := j.UnmarshalJSON([]byte("this is not base64"))
+	assert.Error(t, err)
+
+	b := `
+{
```
Comment (polaroi8d) [inline agent/api/v1/deploy_test.go:215]: I think you missed formatting here.

**Review by polaroi8d — CHANGES_REQUESTED** [review]
Please @clebs read the [CONTRIBUTING.md](https://github.com/dyrector-io/dyrectorio/blob/develop/CONTRIBUTING.md) we have a strict branching name convention (using eg. `refactor(test): add unit tests for agent` for PR title)

**Comment by clebs** [comment]
Hi @polaroi8d,
thanks for the information, I have opened a new PR following the branching and title conventions here:
#257

**Pull request closed without merging**